### ISSUE TITLE
Unknown escape sequence on a technically valid escape sequence

### ISSUE THREAD

**Issue by scrouthtv** [golang/go#43337]
TLDR: `\033[` works as expected, `\e[` yields `unknown escape` although both do the same thing.

### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.15.6 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/lenni/.cache/go-build"
GOENV="/home/lenni/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/lenni/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/lenni/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/lenni/git/easy-copy/src/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build736783570=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?
On Linux ([and recent Windows versions](https://devblogs.microsoft.com/commandline/24-bit-color-in-the-windows-console/)), text on the console can be colored with escape sequences of the form
```
\e   |       | [0-8 21-28] for text attributes       |
\033 | + [ + | [30-39,90-99] for foreground colors   | + m
             | [40-49,100-109] for background colors |
```
*For more information, see https://misc.flogisoft.com/bash/tip_colors_and_formatting and https://en.wikipedia.org/wiki/Escape_sequences_in_C#ref_Note1.*

For example, both `\e[31;44m` and `\033[31;44m` should create red text on blue background.
This does also work in C:
![image](https://user-images.githubusercontent.com/23227405/102966047-61959e80-44ef-11eb-97ea-6654d5ac76a8.png)
<details><pre>
#include <stdio.h>

int main(int argc, char** argv) {
	printf("\e[31;44mHello\e[0m\n");
	printf("--\n");
	printf("\033[31;44mHello again\033[0m\n");
}
</pre></details>

**However**, trying to create string which contains `\e` yields a compiler error: https://play.golang.org/p/5XhK4vANSqI
The very same code with `\033` works can compile and works as expected: https://play.golang.org/p/Wv3sJZBjrVI
![image](https://user-images.githubusercontent.com/23227405/102966219-be915480-44ef-11eb-9507-65b800e161da.png)

*When working with these examples, keep in mind that there are two escape sequences in each line, the one in the front, setting the color, and the one at the end, clearing the color.*

### What did you expect to see?
I would expect both `\e` and `\033` to work the same way as they are used to do the same thing.
If someone is trying to colorize the console output of their tools, and they try to use `\e`, which fails, it feels counterintuitive for `\033` to work.

### What did you see instead?
A string containing `\e` yields a compiler error, whilst a string containing `\033` works as expected.

This may somewhat be related to #11575. I don't know which category this belongs to.

**Comment by ianlancetaylor** [comment]
The escape sequences supported in Go string literals are defined https://golang.org/ref/spec#Rune_literals.  The escape sequence `\e` is not supported by Go.

It is also not supported by standard C, although it is a common addition.  This is mentioned on the Wikipedia page you cite, at https://en.wikipedia.org/wiki/Escape_sequences_in_C#Non-standard_escape_sequences .

Closing because this is not a bug.
